test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -5746,6 +5746,29 @@ public async Task ValidateExtensionLifecycleLogs()
             JObject.Parse(configJson);
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task TestStoreInputsInOrchestrationHistory()
+        {
+            DurableTaskOptions options = new DurableTaskOptions();
+            options.StoreInputsInOrchestrationHistory = true;
+
+            using (var host = TestHelpers.GetJobHostWithOptions(
+                   this.loggerProvider,
+                   options))
+            {
+                await host.StartAsync();
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "foo", this.output);
+                await client.WaitForCompletionAsync(this.output);
+                var status = await client.InnerClient.GetStatusAsync(client.InstanceId, showHistory: true, showHistoryInput: true);
+
+                var input = status.History[1].Value<string>("Input");
+                Assert.Equal("[\"foo\"]", input);

```

Comment:
Just to be a little more thorough we should add another activity to this test that doesn't take an input and make sure Input is still returned in the GetStatusAsync result as null. You might need to create a new orchestrator if you can't find one with two activities like this.